doc/user/content/transform-data/idiomatic-materialize-sql/_index.md

```diff
@@ -0,0 +1,104 @@
+---
```

Comment:
For now, creating a new section.

Made a new section for this content as the current Patterns section -- other than the top-k and windows functions pages (which have been re-written and moved  to the idiomatic section) -- seem more like guides/patterns but I will have a better sense as I work through that section in later work.